### ISSUE TITLE
Backport missing #include <cstdint> for upcomit gcc-13 to stable branch

### ISSUE THREAD

**Issue by trofi** [llvm/llvm-project#55711]
Without the change llvm build fails on this week's gcc-13 snapshot as:

    [  0%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o
    In file included from llvm/lib/Support/Signals.cpp:14:
    llvm/include/llvm/Support/Signals.h:119:8: error: variable or field 'CleanupOnSignal' declared void
      119 |   void CleanupOnSignal(uintptr_t Context);
          |        ^~~~~~~~~~~~~~~

    [ 91%] Building CXX object unittests/Support/CMakeFiles/SupportTests.dir/Base64Test.cpp.o
    In file included from llvm/unittests/Support/Base64Test.cpp:14:
    llvm/include/llvm/Support/Base64.h: In function 'std::string llvm::encodeBase64(const InputBytes&)':
    llvm/include/llvm/Support/Base64.h:29:5: error: 'uint32_t' was not declared in this scope
       29 |     uint32_t x = ((unsigned char)Bytes[i] << 16) |
          |     ^~~~~~~~

Two trivial commits should be enough: ff1681ddb303223973653f7f5f3f3435b48a1983 5e9be93566f39ee6cecd579401e453eccfbe81e5

/cc @marxin

**Comment by trofi** [comment]
/cherry-pick ff1681ddb303223973653f7f5f3f3435b48a1983 5e9be93566f39ee6cecd579401e453eccfbe81e5

**Comment by trofi** [comment]
No, I pushed them directly as looking trivial. Should I revert them and upload on phabricator?

**Comment by tstellar** [comment]
I like to get a second opinion usually before I backport patches, do you know  why this change was needed?

**Comment by thesamesam** [comment]
> I like to get a second opinion usually before I backport patches, do you know why this change was needed?

It fixes building with GCC 13 (currently in development).

**Comment by tstellar** [comment]
@thesamesam Ok, thank you.

**Comment by trofi** [comment]
> I like to get a second opinion usually before I backport patches, do you know why this change was needed?

In weekly gcc-13 shapshot `<string>` header stopped including `<cstdint>` (intended change: https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=6f038efd93593da6e661b829d1bd3877e75550f1) for better headers hygiene. As a result some packages started to fail.

**Comment by tru** [comment]
This one have some test failures in the bots here: https://github.com/llvm/llvm-project-release-prs/runs/6720374584?check_suite_focus=true

```
Failed Tests (2):
  LLVM-Unit :: Support/./SupportTests.exe/FormatVariadicTest.BigTest
  LLVM-Unit :: Support/./SupportTests.exe/NativeFormatTest.BoundaryTests
```

Unsure if they are related. Can you have a look @trofi or @thesamesam

**Comment by trofi** [comment]
> This one have some test failures in the bots here: https://github.com/llvm/llvm-project-release-prs/runs/6720374584?check_suite_focus=true
> 
> ```
> Failed Tests (2):
>   LLVM-Unit :: Support/./SupportTests.exe/FormatVariadicTest.BigTest
>   LLVM-Unit :: Support/./SupportTests.exe/NativeFormatTest.BoundaryTests
> ```
> 
> Unsure if they are related. Can you have a look @trofi or @thesamesam

I'd say it's unrelated. Looking at a random other PR against this repository we see the same failures already happening. For example any of these two:
- https://github.com/llvm/llvm-project-release-prs/runs/6603412835?check_suite_focus=true
- https://github.com/llvm/llvm-project-release-prs/runs/6719881582?check_suite_focus=true

Some test detail: `Support/./SupportTests.exe/NativeFormatTest.BoundaryTests` tests if   `EXPECT_EQ("INF", format_number(std::numeric_limits<double>::infinity(), FloatStyle::Fixed));` returns "INF". It returns `"inf"` for some reason. I can't see why exactly. Possibly due to some very unusual overload.

**Comment by tstellar** [comment]
Merged: 4d039a7a71899038b3bc6ed6fe5a8a48d915caa0 a517f3439671f18327ada0f5d8ca328934cc337e
